doc/specs/vulkan/chapters/fundamentals.txt

```diff
@@ -284,10 +284,10 @@ frees.
 It is an application's responsibility to track the lifetime of Vulkan
 objects, and not to destroy them while they are still in use.
 
-Application-owned memory is immediately consumed by any Vulkan command it is
-passed into.
+Application-owned memory is borrowed by any Vulkan command it is passed into
```

Comment:
I think "borrowed" makes more sense, but I am concerned we've not used this term before. Would feel happier if you could put it as a glossary entry as well?

There's some notion of this concept being needed more widely in the API, since lifetimes are a bit tangled at the moment. For instance, ideally I'd want to be able to tag parameters for each call saying whether they are borrowed during the call, and whether they are released or not, and for later calls, when stored parameters are released. Similar to the externsync stuff, I guess. Potentially that would make the whole "command pools are externsync during command recording" easier to express as well.

I like "borrowed" since currently we don't have a term for it.

Reply:
(I don't know if "borrowed" is really the opposite to "released", just winging it here :P)

Reply:
I dislike "borrowed", it implies a temporary change in ownership which doesn't exist.

Reply:
@TobiasHector I guess it does, if by "released" you mean something like "return borrowed item". It was intended for pass by reference (pointer) parameters, but I suppose it would work for other things too (handles). It does not necessarily implies `externsync`, but whether the parameter is borrowed and ignored, passed as const, or as mutable is a strong indication.

@jeffbolznv How so? That's exactly what it is. You no longer own the object (i.e. you have no moral right to free it while someone else owns it). Definition of "ownership" is not needed, because "borrowed" would be defined explicitly anyway. I stole it from [Rust here](https://doc.rust-lang.org/1.8.0/book/references-and-borrowing.html) if anyone wonders.
Anyway, I dislike "consumed" (it means `std::move()` in my dictionary) and I fail to find other precedent for this term on the internet.

Reply:
I guess we're talking about ownership here, and actually the spec does call out "acquire" and "release" as verbs for this. So you could say that calling the command "acquires ownership of the pointer" and then when the command returns "ownership is released". 

There's possibly some question here with regards externsync and const pointers about whether they're fully owned (i.e. might be written, don't even read them during that call) or if there's shared ownership (i.e. only read, app can read at the same time).

Reply:
Rewrote this trying "acquire ownership" term.